File: Hash Table/349_Intersection-of-Two-Arrays/349_Intersection-of-Two-Arrays.cpp
/******************************************************************************
 * Title:
 *     349. Intersection of Two Arrays
 *     349. 两个数组的交集
 * Address:
 *     https://leetcode-cn.com/problems/intersection-of-two-arrays/
******************************************************************************/

#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

// 方法一：哈希表
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> hashSet1( nums1.begin(), nums1.end() );
        unordered_set<int> hashSet2( nums2.begin(), nums2.end() );

        vector<int> ans;
        for ( auto iter = hashSet1.begin(); iter != hashSet1.end(); ++iter ) {
            if ( hashSet2.count( *iter ) )
                ans.push_back( *iter );
        }

        return ans;
    }
};

int main()
{
    

    return 0;
}

File: Hash Table/349_Intersection-of-Two-Arrays/349_Intersection-of-Two-Arrays.py
"""
 * Title:
 *     349. Intersection of Two Arrays
 *     349. 两个数组的交集
 * Address:
 *     https://leetcode-cn.com/problems/intersection-of-two-arrays/
"""

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1 = set(nums1)
        set2 = set(nums2)

        ans = list()
        for s1 in set1:
            if s1 in set2:
                ans.append(s1)
        
        return ans